File: src/plugin/combat/magic/modern/teleport/FaladorTeleportSpell.java
package plugin.combat.magic.modern.teleport;

import org.redrune.game.global.WorldTile;
import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/27/2017
 */
public class FaladorTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 37;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(WATER_RUNE, 1, AIR_RUNE, 3, LAW_RUNE, 1);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2964, 3379, 0);
	}
	
	@Override
	public int spellId() {
		return 46;
	}
	
	@Override
	public double exp() {
		return 48;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.REGULAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/modern/teleport/FaladorTeleportSpell.kt
package plugin.interaction.combat.magic.modern.teleport

import org.redrune.game.content.plugin.combat.spell.type.TeleportSpellPlugin
import org.redrune.game.global.WorldTile
import org.redrune.utility.constants.MagicConstants
import org.redrune.utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 7/27/2017
 */
class FaladorTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 37
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.WATER_RUNE, 1, MagicConstants.AIR_RUNE, 3, MagicConstants.LAW_RUNE, 1)
    }

    override fun destination(): WorldTile {
        return WorldTile(2964, 3379, 0)
    }

    override fun spellId(): Int {
        return 46
    }

    override fun exp(): Double {
        return 48.0
    }

    override fun book(): MagicBook {
        return MagicBook.REGULAR
    }
}